.github/workflows/simple-build.yml

```
@@ -34,7 +37,13 @@ jobs:
       CXXFLAGS: ${{ matrix.toolchain.CXXFLAGS }}
 
     steps:
-      - name: check out
-        uses: actions/checkout@v1
+      - name: checkout (push)
+        if: ${{ github.event_name == 'push' }}
+        uses: actions/checkout@v2
+      - name: checkout (pull request)
+        if: ${{ github.event_name == 'pull_request' }}
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
       - name: build
         run: '"${CXX}" ${CXXFLAGS} -o quick-lint-js -I src -I vendor/boost src/*.cpp vendor/boost/libs/container/src/*.cpp vendor/boost/libs/container/src/alloc_lib.c'
```

Nit: Let's separate the checkout from the important work with a blank line (here and in the other workflows):

```suggestion
        with:
          ref: ${{ github.event.pull_request.head.sha }}

      - name: build
        run: '"${CXX}" ${CXXFLAGS} -o quick-lint-js -I src -I vendor/boost src/*.cpp vendor/boost/libs/container/src/*.cpp vendor/boost/libs/container/src/alloc_lib.c'
```